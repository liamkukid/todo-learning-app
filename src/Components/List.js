import PropTypes from 'prop-types';
import style from './list.module.scss';

export default function List({ todos, onDone }) {
  const handleChange = e => {
    onDone(e.target.value, e.target.checked);
  };

  if (todos && todos.length > 0) {
    return (
      <div className={style.list}>
        {todos.map(todo => (
          <div
            key={todo.id}
            className={`${style.list_item} ${
              todo.done ? style.list_item_done : ''
            }`}
          >
            <input
              type="checkbox"
              value={todo.id}
              checked={todo.done}
              onChange={handleChange}
            />
            {todo.title}
          </div>
        ))}
      </div>
    );
  }
}

List.propTypes = {
  todos: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      done: PropTypes.bool,
      id: PropTypes.string,
    })
  ),
  onDone: PropTypes.func.isRequired,
};

List.defaultProps = {
  todos: [],
};
