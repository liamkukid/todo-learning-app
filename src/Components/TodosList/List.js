import { useSelector } from 'react-redux';

import TodoDoneCheckbox from './TodoDoneCheckbox';
import TodoRemoveButton from './TodoRemoveButton';
import style from './list.module.scss';
import { filteredTodos } from '../../features/selectTodos';

export default function List() {
  const todos = useSelector(filteredTodos);

  if (!todos || todos?.length <= 0) {
    return <div />;
  }

  return (
    <div className={style.list}>
      {todos.map(todo => (
        <div
          key={todo.id}
          className={`${style.list_item} ${
            todo.done ? style.list_item_done : ''
          }`}
        >
          <TodoDoneCheckbox todo={todo} />
          {todo.title}
          <TodoRemoveButton todoId={todo.id} />
        </div>
      ))}
    </div>
  );
}
