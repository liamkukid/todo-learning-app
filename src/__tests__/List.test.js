import renderer from 'react-test-renderer';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import todosReducer from '../features/todosSlice';
import List from '../Components/TodosList/List';

it('renders correctly with no todos', () => {
  const store = configureStore({
    reducer: {
      todos: todosReducer,
    },
    preloadedState: {
      todos: [],
    },
  });
  const tree = renderer
    .create(
      <Provider store={store}>
        <List />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with todos', () => {
  const store = configureStore({
    reducer: {
      todos: todosReducer,
    },
    preloadedState: {
      todos: [
        {
          title: 'todo 1',
          done: true,
          id: 'todo_1_id',
        },
        {
          title: 'todo 2',
          done: false,
          id: 'todo_2_id',
        },
      ],
    },
  });
  const tree = renderer
    .create(
      <Provider store={store}>
        <List />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
