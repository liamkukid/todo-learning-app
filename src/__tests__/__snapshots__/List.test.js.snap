// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with no todos 1`] = `<div />`;

exports[`renders correctly with todos 1`] = `
<div
  className="list"
>
  <div
    className="list_item list_item_done"
  >
    <input
      checked={true}
      data-testid="todo-checkbox"
      onChange={[Function]}
      type="checkbox"
      value="todo_1_id"
    />
    todo 1
    <button
      className="remove_todo_button"
      onClick={[Function]}
      type="button"
      value="todo_1_id"
    >
      <img
        alt="gabarge"
        className="gabarge-logo"
        src="gabarge.svg"
      />
    </button>
  </div>
  <div
    className="list_item "
  >
    <input
      checked={false}
      data-testid="todo-checkbox"
      onChange={[Function]}
      type="checkbox"
      value="todo_2_id"
    />
    todo 2
    <button
      className="remove_todo_button"
      onClick={[Function]}
      type="button"
      value="todo_2_id"
    >
      <img
        alt="gabarge"
        className="gabarge-logo"
        src="gabarge.svg"
      />
    </button>
  </div>
</div>
`;
